src/rantanplan/rhymes.py

```diff
@@ -0,0 +1,190 @@
+    word of each line (stressed_endings), return a tuple with two lists:
+    - rhyme pattern of each line (e.g., a, b, b, a)
+    - rhyme ending of each line (e.g., ado, ón, ado, ón)
+    The rhyme checking method can be assonant (assonance=True) or
+    consonant (default). Moreover, some dipthongs relaxing rules can be
+    applied (relaxation=False) so the weak vowels are removed when checking
+    the ending syllables.
+    By default, all verses are checked, that means that a poem might match
+    lines 1 and 100 if the ending is the same. To control how many lines
+    should a matching rhyme occur, an offset can be set to an arbitrary
+    number, effectively allowing rhymes that only occur between
+    lines i and i + offset. The symbol for free verse can be set
+    using free_verse_symbol (defaults to '-')"""
+    codes = {}
+    code_numbers = []
+    unique = set()
+    # Clean consonants as needed and assign numeric codes
+    for stressed_ending, _, stressed_position in stressed_endings:
+        stressed_ending_upper = stressed_ending[stressed_position].upper()
+        stressed_ending[stressed_position] = stressed_ending_upper
+        if relaxation:
+            ending = "".join(WEAK_VOWELS_RE.sub(r"\1", syll, count=1)
+                             for syll in stressed_ending)
+        else:
+            ending = "".join(stressed_ending)
+        ending = DIPHTHONG_Y_RE.sub(r"\1i\2", ending)
+        if assonance:
+            ending = CONSONANTS_RE.sub(r"", ending)
+        else:
+            # Consonance
+            ending = DIPHTHONG_H_RE.sub(r"\1\2", ending)
+            ending = INITIAL_CONSONANTS_RE.sub(r"", ending, count=1)
+        ending = strip_accents(ending)
+        if ending not in codes:
+            codes[ending] = len(codes)
+            unique.add(codes[ending])
+        else:
+            unique.discard(codes[ending])
+        code_numbers.append(codes[ending])
+    code2endings = {v: k for k, v in codes.items()}
+    # Adjust for free verses and assign letter codes
+    letters = {}
+    rhymes = []
+    endings = []
+    last_found = {}
+    for index, rhyme in enumerate(code_numbers):
+        if rhyme in unique:
+            rhyme_letter = -1  # free verse
+            endings.append("")
+        else:
+            if rhyme not in letters:
+                letters[rhyme] = len(letters)
+            rhyme_letter = letters[rhyme]
+            # Reassign free verses if an offset is set
+            if (rhyme in last_found
+                    and offset is not None
+                    and index - last_found[rhyme] > offset):
+                rhymes[last_found[rhyme]] = -1  # free verse
+                endings[last_found[rhyme]] = ''
+            last_found[rhyme] = index
+            endings.append(code2endings[rhyme])
+        rhymes.append(rhyme_letter)
+    # Reorder letters
+    rhyme_letters = []
+    letters = {}
+    stresses = []
+    for rhyme in rhymes:
+        if rhyme < 0:  # free verse
```

Comment:
I think the correct term is 'blank verse'. 'Free verses' are verses without a regular metre (prose)

Reply:
I'll double check and change accordingly.